crates/mpt/src/test_utils.rs

```diff
@@ -118,15 +118,62 @@ pub(crate) async fn get_live_derivable_transactions_list(
     Ok((root, preimages, consensus_txs))
 }
 
+/// The default, no-op implementation of the [TrieProvider] trait, used for testing.
+#[derive(Debug, Clone, Copy)]
+pub struct NoopTrieProvider;
+
+impl TrieProvider for NoopTrieProvider {
+    type Error = String;
+
+    fn trie_node_preimage(&self, _key: B256) -> Result<Bytes, Self::Error> {
+        Ok(Bytes::new())
+    }
+
+    fn bytecode_by_hash(&self, _code_hash: B256) -> Result<Bytes, Self::Error> {
+        Ok(Bytes::new())
+    }
+
+    fn header_by_hash(&self, _hash: B256) -> Result<Header, Self::Error> {
+        Ok(Header::default())
+    }
+}
+
+/// The default, no-op implementation of the [TrieHinter] trait, used for testing.
+#[derive(Debug, Clone, Copy)]
+pub struct NoopTrieHinter;
```

Comment:
I don't think these belong in test utils unfortunately. Succinct and other downstream implementers use this - non FPVM targets don't have hints.

Reply:
Specifically the hinter - if we really want to we can keep the provider behind the test utils, but may as well also distribute it if we have the noop hinter distributed.